src/wagi_server.rs

```diff
@@ -65,9 +71,10 @@ impl WagiServer {
                         }))
                     })
                 });
-                Server::builder(tls::TlsHyperAcceptor::new(&self.address, &tls.cert_path, &tls.key_path).await?)
-                    .serve(mk_svc)
-                    .await?;
+                let server = Server::builder(tls::TlsHyperAcceptor::new(&self.address, &tls.cert_path, &tls.key_path).await?)
+                    .serve(mk_svc);
+                let graceful = server.with_graceful_shutdown(shutdown_signal());
```

Comment:
We had problems with `with_graceful_shutdown` in Spin because it kept the process alive until all connections had closed, which could be a couple of minutes due to keep-alives.  See https://github.com/fermyon/spin/issues/73 for the problem and links to relevant Hyper issues, and https://github.com/fermyon/spin/pull/232 for how we solved it there.